integreat_cms/api/v3/events.py

```diff
@@ -193,8 +193,12 @@ def events(request, region_slug, language_slug):
     now = timezone.now().date()
     for event in region.events.prefetch_public_translations().filter(archived=False):
         if event_translation := event.get_public_translation(language_slug):
+            event_is_not_recurring = not event.recurrence_rule
+            recurring_event_should_be_appended_once = "combine_recurring" in request.GET
             # Either it's in the future or it's recurring and the last recurrence is >= now
-            if not event.is_past:
+            if not event.is_past and (
+                event_is_not_recurring or recurring_event_should_be_appended_once
+            ):
                 result.append(transform_event_translation(event_translation))
             if "combine_recurring" not in request.GET:
                 for future_event in transform_event_recurrences(event_translation, now):
```

Comment:
I think we could simplify the nested structure of this loop a bit:
```
    for event in region.events.prefetch_public_translations().filter(archived=False):
        if not event.is_past and (
            event_translation := event.get_public_translation(language_slug)
        ):
            combine_recurring_events = "combine_recurring" in request.GET
            if event.is_recurring and not combine_recurring_events:
                result.extend(iter(transform_event_recurrences(event_translation, now)))
            else:
                result.append(transform_event_translation(event_translation))
```

Reply:
done